api-consumption/src/main/java/org/zalando/nakadi/service/subscription/state/StartingState.java

```diff
@@ -85,11 +57,52 @@ private void registerSessionAndStartStreaming() {
 
         try {
             getContext().registerSession();
+            if (!areStreamingSlotsAvailable(
+                    getZk().listSessions().stream()
+                            .filter(s -> !s.getId().equals(getSessionId()))
+                            .collect(Collectors.toList()))) {
+                return;
+            }
         } catch (Exception ex) {
             switchState(new CleanupState(ex));
             return;
         }
+
         switchState(new StreamingState());
     }
 
+    private boolean areStreamingSlotsAvailable(final Collection<Session> sessions) {
```

Comment:
`nitpick`: While looking closer to this method, It seems to be not the best name for the method `areStreamingSlotsAvailable` is right now responsible for not only checking, but also for switching state of streaming to a CleanupState (with reason). 

Maybe it makes sense either to rename the method or to move switchToState somehow to a caller part (don't know what is the best name here). Overall - there is no bug, but looks suspicious, feel free to leave it as it is.